### ISSUE TITLE
[Feature] Making the sidebar category accessible for keyboard users

### ISSUE THREAD

**Issue by IbenTesara** [ng-doc/ng-doc#257]
### Description

Currently the `NgDocSidebarCategoryComponent` has an inaccessible button with the class `ng-doc-sidebar-category-button`. 

As we use NgDoc for our open-source packages where accessibility and WCAG/WAI-ARIA compliance is at the forefront, having a documentation website that is then not accessible becomes a problem.

This button is inaccessible because of the fact that this is a div used as a button, instead of using a button. There are two possible solutions, either replacing this div with a button or in case that would cause too many issues with the styling, adding the required properties.

I'm willing to open a PR for this as this is a relatively small feature, but I would appreciate some input on whether you prefer to keep the div approach instead of the button approach. The button approach is obviously the more semantic correct one, but I can understand the preference to keep the div approach as well.

### Proposed solution

Replace:
`<div class="ng-doc-sidebar-category-button" (click)="toggle()"> ... </div>`

With:
`<button class="ng-doc-sidebar-category-button" (click)="toggle()"> ... </button>`

### Alternatives considered

Replace:
`<div class="ng-doc-sidebar-category-button" (click)="toggle()"> ... </div>`

With:
`<div class="ng-doc-sidebar-category-button" role="button" tabindex="0" (keydown.enter)="toggle()" (click)="toggle()"> ... </div>`
